app/services/zendesk_sender.rb

```diff
@@ -1,17 +1,28 @@
 class ZendeskSender
   attr_accessor :ticket_payload
 
-  class << self
-    def send!(ticket_payload)
-      ZendeskSender.new(ticket_payload).send!
-    end
+  def self.call(ticket_payload)
+    new(ticket_payload).call
   end
 
   def initialize(ticket_payload)
     @ticket_payload = ticket_payload
+    @return_hash = {}
```

Comment:
Does '@return_hash' need to be an instance variable? Could it just be a local variable inside the call method? And then it could just be passed to catch_error. I am asking this to check my understanding of when to use an instance variable as I am not entirely sure myself.

Reply:
My logic was just that having it be an instance variable is a more elegant solution than having to have it be passed to the catch_error, and then catch_error would need to return it back so that call could return it.

That said, looking at it again has made me think of what I think is a nicer way of doing it. I've amended it onto the most recent commit, what do you think?

Reply:
Oh I like that @AlexRyanShep. I would be interested in other peoples thoughts but it feels more readable having it as a whole hash inside the method - where you can see all the values at once. And more encapsulated perhaps!?!?